rts/lra/lra-test/lra-test-arquillian-extension/src/main/java/io/narayana/lra/arquillian/spi/LRAAnnotationAdjuster.java

```diff
@@ -95,6 +97,25 @@ static void adjustLRAAnnotation(Class<?> clazzToLookFor, LRA originalLRAAnnotati
                 throw new IllegalStateException("Cannot change annotation " + originalLRAAnnotation
                         + " of class " + clazzToLookFor + " in JDK8 way", e);
             }
+        } else if (doesJDKDefineFieldName(ANNOTATION_CACHE_FIELD_NAME)) {
+            // OpenJ9
+            try {
+                Field cacheField = Class.class.getDeclaredField(ANNOTATION_CACHE_FIELD_NAME);
+                cacheField.setAccessible(true);
+                Object cache = cacheField.get(clazzToLookFor);
+
+                Class<? extends Object> cacheClass = cache.getClass();
+                Field mapField = cacheClass.getDeclaredField(ANNOTATION_MAP_FIELD_NAME);
+                mapField.setAccessible(true);
+                Map<Class<? extends Annotation>, Annotation> annotationMap = (Map<Class<? extends Annotation>, Annotation>) mapField.get(cache);
+                annotationMap.put(LRA.class, new LRAWrapped(originalLRAAnnotation));
+            } catch (Exception e) {
+                throw new IllegalStateException("Cannot change annotation " + originalLRAAnnotation
+                        + " of class " + clazzToLookFor + " in JDK8 way", e);
```

Comment:
probably mention that this is OpenJ9 so the message is different?

Reply:
+1, thanks a lot for spotting this